nexus/db-queries/src/db/datastore/sled.rs

```diff
@@ -82,7 +87,12 @@ impl DataStore {
                         &sled_update.id().to_string(),
                     ),
                 )
-            })
+            })?;
+
+        // We compare only seconds since the epoch, because writing to and
+        // reading from the database causes us to lose precision.
+        let was_modified = now.timestamp() == sled.time_modified().timestamp();
+        Ok((sled, was_modified))
```

Comment:
This seems probably alright but not great.  This would produce false positives if the value was recently modified and we _didn't_ change it.  That's not a problem for the current caller -- it'll just trigger an extra inventory collection, which isn't that big a deal.  I'm more worried someone will look at the docs and rely on the boolean being correct for correctness (whereas right now it's just an optimization).

I _think_ in this PR there's only one caller of this function that needs this behavior, but a lot of the delta here is around changing the existing callers to ignore this second boolean.

So what do you think of this: don't change this function at all, and have that one caller check the modification time on the returned sled?  That way the limitation is specific to that one caller and the general code is not providing a boolean that might not be accurate.  (edit: oh, all callers would need to do the `sled_update.into_insertable()` themselves to do this.  That still seems better to me but definitely more debatable.)

Also I think the call that triggers this function only happens when a sled agent starts up, right?  It doesn't seem that bad to just always trigger a collection in that case.

Reply:
I had a couple similar comments staged but I forgot to hit "submit": one complained that this was icky and wondered if we could get an accurate answer from the database instead of having to check `time_modified` as a proxy, and the other was a request to not use `bool` because it's not obvious to me what it means at a call site like this:

```rust
let (_, was_upserted) = sled_upsert(...)?;
if was_upserted {
    // was the sled updated or inserted?
}
```

Your suggestion would address both of my concerns.

Reply:
I hate the bool in a tuple return value, but I also didn't want to force all callers to call `into_insertable`. That seems to make the `SledUpdate` type somewhat of a weird intermediate state, and no longer part of the API. We only turn it into an insertable to use the magic id stuff and grab the mod time, etc.... I also find that whole [macro transformation](https://github.com/oxidecomputer/omicron/blob/main/nexus/db-macros/src/lib.rs#L182) to be way too much magic for my liking, but changing that is outside the scope of this PR.

The other alternative I thought of when writing this code was to pass in an optional timestamp and set it in `sled_update`, but that seems nearly equivalent to the bool, except now it can't be ignored by the caller. They have to pass in a `None`, which I find more distasteful than returning a bool. 

All this is how I ended up where I did, and a other than changing the return value to a struct or adding a newtype wrapper around the bool with a doc comment, I don't think there's a better way forward.

Reply:
In light of my counter arguments, and moving onto more critical things, I'm going to keep this as is. We can always revisit.